destination/src/main/scala/quasar/plugin/sqlserver/destination/package.scala

```diff
@@ -142,10 +145,68 @@ package object destination {
         statement.addBatch(sql.toString)
         prefix.setLength(length)
       }
-
       statement.executeBatch()
     }
 
     HC.createStatement(batch).void
   }
+
+  def pathFragment[F[_]: Applicative: MonadResourceErr](scheme: String, path: ResourcePath)
+      : F[(Fragment, String, Option[String])] = {
+    val actualPath = path./:(ResourceName(scheme))
+    val objF = jdbc.resourcePathRef(actualPath) match {
+      case None => MonadResourceErr[F].raiseError(ResourceError.notAResource(actualPath))
+      case Some(obj) => obj.pure[F]
+    }
+    objF map {
+      case Left(t) =>
+        val hy = SQLServerHygiene.hygienicIdent(t)
+        (Fragment.const0(hy.forSqlName), hy.unsafeForSqlName, None)
+      case Right((d, t)) =>
+        val hyD = SQLServerHygiene.hygienicIdent(d)
+        val hyT = SQLServerHygiene.hygienicIdent(t)
+        (Fragment.const0(hyD.forSqlName) ++ fr0"." ++ Fragment.const0(hyT.forSqlName),
+          hyT.unsafeForSqlName,
+          hyD.unsafeForSqlName.some)
+    }
+  }
+
+  def startLoad(logHandler: LogHandler)(
+    writeMode: WriteMode,
+    obj: Fragment,
+    unsafeName: String,
+    unsafeSchema: Option[String],
+    columns: NonEmptyList[(HI, SQLServerType)],
+    idColumn: Option[Column[_]])
+      : ConnectionIO[Unit] = {
+    val prepareTable = writeMode match {
+      case WriteMode.Create => createTable(logHandler)(obj, columns)
+      case WriteMode.Replace => replaceTable(logHandler)(obj, columns)
+      case WriteMode.Truncate => truncateTable(logHandler)(obj, unsafeName, unsafeSchema, columns)
+      case WriteMode.Append => appendToTable(logHandler)(obj, unsafeName, unsafeSchema, columns)
+    }
+    val mbCreateIndex = idColumn traverse_ { col =>
+      val colFragment = Fragments.parentheses(Fragment.const(SQLServerHygiene.hygienicIdent(Ident(col.name)).forSqlName))
+      createIndex(logHandler)(obj, unsafeName, colFragment)
+    }
+    prepareTable >> mbCreateIndex
+  }
+
+  def hygienicColumns[F[_]: Functor, A](cols: F[Column[A]]): F[(HI, A)] = cols map { c =>
+    (SQLServerHygiene.hygienicIdent(Ident(c.name)), c.tpe)
+  }
+
+  def indexName(unsafeName: String): String =
+    s"precog_id_idx_$unsafeName"
+
+  def createIndex(log: LogHandler)(obj: Fragment, unsafeName: String, column: Fragment): ConnectionIO[Int] = {
+    val strName = s"precog_id_idx_$unsafeName"
```

Comment:
I think we want to use `indexName` here.

Reply:
Thank you!